src/features/callAssignments/models/CallAssignmentModel.ts

```diff
@@ -93,6 +95,35 @@ export default class CallAssignmentModel {
     return data.target.filter_spec?.length === 0;
   }
 
+  setCooldown(cooldown: number) {
+    const state = this._store.getState();
+    const callAssignment = state.callAssignments.callAssignments.find(
+      (ca) => ca.id == this._id
+    );
+
+    if (callAssignment) {
+      this._store.dispatch(callAssignmentLoad());
+      const promise = fetch(
```

Comment:
The need for some sort of API client adapter is starting to emerge.

Reply:
i liked the idea of injecting the needed adapters to the class, it felt versatile to me (on a superficial level of understanding it)